paasta_tools/utils.py

```diff
@@ -2527,6 +2528,30 @@ def get_git_repo_config(self, repo_name: str) -> Dict:
     def get_hpa_always_uses_external_for_signalfx(self) -> bool:
         return self.config_dict.get("hpa_always_uses_external_for_signalfx", False)
 
+    def get_legacy_autoscaling_signalflow(self) -> str:
+        # This Signaflow attempts to recreate the behavior of the legacy autoscaler,
+        # which averages the number of instances needed to handle the current (or
+        # averaged) load instead of the load itself. This leads to more stable behavior.
+        # Note that it returns the percentage by which we want to scale , so its target
+        # in the HPA should always be 1.
+        # See PAASTA-16756 for details.
+        default_legacy_autoscaling_signalflow = """offset = {offset}
+setpoint = {setpoint}
+moving_average_window = '{moving_average_window_seconds}s'
+filters = filter('paasta_service', '{paasta_service}') and filter('paasta_instance', '{paasta_instance}') and filter('paasta_cluster', '{paasta_cluster}')
+
+current_replicas = data('kube_deployment_spec_replicas', filter=filters, extrapolation="last_value").sum(by=['paasta_cluster'])
+load_per_instance = data('{signalfx_metric_name}', filter=filters, extrapolation="last_value", maxExtrapolations=10).below(1, clamp=True)
+
+desired_instances_at_each_point_in_time = (load_per_instance - offset).sum() / (setpoint - offset)
+desired_instances = desired_instances_at_each_point_in_time.mean(over=moving_average_window)
+
+(desired_instances / current_replicas).above(0).publish()
+"""
```

Comment:
Just to be sure -- the only difference here is that we're using `kube_deployment_spec_replicas` instead of `kube_hpa_status_current_replicas`, yeah?

Reply:
yep, that's the only thing I changed on the query